src/blockchain.rs

```diff
@@ -217,86 +219,131 @@ impl Blockchain {
             + 1
             != block.id()
         {
+            println!("ID NOT INCREMENTED PROPERLY");
             return false;
         }
 
         // TODO -- include checks on difficulty and paysplit here to validate
 
         // Validate the burnfee
         let previous_block = self.fork_tree.block_by_hash(previous_block_hash).unwrap();
-        if block.burnfee()
-            != BurnFee::burn_fee_calculation(
-                previous_block.burnfee() as f64,
+        if block.start_burnfee()
+            != BurnFee::burn_fee_adjustment_calculation(
+                previous_block.start_burnfee() as f64,
                 block.timestamp(),
                 previous_block.timestamp(),
-            )
+            ) as f64
```

Comment:
The previous code makes more sense to me, I still don't see why we need start_burnfee...